security/medusa/l4-constable/chardev.c

```diff
@@ -41,20 +41,18 @@
 #include <linux/jiffies.h>
 #include <linux/sched/task.h>
 #include <linux/uaccess.h>
-#include <linux/slab.h>
 
 #include <linux/medusa/l3/arch.h>
 #include <linux/medusa/l3/registry.h>
 #include <linux/medusa/l3/server.h>
 #include <linux/medusa/l4/comm.h>
 
 #include "teleport.h"
+#include "med_cache.h"
 
 #define MEDUSA_MAJOR 111
 #define MODULENAME "chardev/linux"
 #define CURRENTPTR current
```

Comment:
I know this is not related, but isn't it simpler to use just ``current`` instead of ``CURRENTPTR`` ? I mean I know that if it would get changed in the kernel, we would have to change it in our code only in one place, but we use ``current`` a lot in L2 so there will be many changes anyway (and we can use ``sed`` if something like this happens).

Reply:
I'll remove it. It's not even used consistently.

Reply:
Please checkout also the use of ``MODULENAME`` and ``MEDUSA_MAJOR``. I am not sure but I think that the ``MODULENAME`` was used before with ``MED_PRINTF``, but now this is superseded with ``med_pr_*``, which includes the module name in it by default and therefore should be no longer needed.

Reply:
These are for the character device.